### PR TITLE
Refactor Note to match other classes

### PR DESCRIPTION
**Pull request by aliciamichellew**

### CODE REVIEW

File: src/main/java/seedu/address/model/NoteCatalogue.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+     * The Note must not already exist in the note list.
+     */
+    public void addNote(Note note) {
+        notes.add(note);
+    }
+
+    /**
+     * Replaces the given LocalCourse {@code target} in the list with {@code editedLocalCourse}.
+     * {@code target} must exist in the LocalCourseCatalogue.
+     * The LocalCourse identity of {@code editedLocalCourse} must not be the same as another existing LocalCourse.
```
Comment (lamchenghou) [inline src/main/java/seedu/address/model/NoteCatalogue.java:81]: Outdated javadocs.

File: src/main/java/seedu/address/model/ReadOnlyNoteCatalogue.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import seedu.address.model.notes.Note;
+
+/**
+ * Unmodifiable view of note catalogue.
+ */
+public interface ReadOnlyNoteCatalogue {
+
+    /**
+     * Returns an unmodifiable view of the local course list.
+     * This list will not contain any duplicate local courses.
```
Comment (lamchenghou) [inline src/main/java/seedu/address/model/ReadOnlyNoteCatalogue.java:13]: Nit: Outdated javadocs.

**Review by lamchenghou — APPROVED** [review]
Great job :) Can you help me implement a few bug fixes I've highlighted, thanks!

File: src/main/java/seedu/address/model/notes/UniqueNoteList.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+ * A note is considered unique by comparing using {@code Note#isSameNote(Note)}. As such, adding and updating of
+ * persons uses Note#isSameNote(Note) for equality to ensure that the person being added or updated is
+ * unique in terms of identity in the UniqueNoteList. However, the removal of a note uses Note#equals(Object)
+ * to ensure that the note with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Note#isSameNote(Note)
+ */
+public class UniqueNoteList implements Iterable<Note> {
```
[the remark this answers is not in the document]
Comment (aliciamichellew) [reply src/main/java/seedu/address/model/notes/UniqueNoteList.java:26]: I'm not sure how much refactoring I need to implement indexing. Can we leave this like this first? We'll look into it later on. Another way I can think of without indexing is just using the `equals` function in Java directly. What do you think?

File: src/main/java/seedu/address/storage/JsonSerializableNoteCatalogue.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+     *
+     * @throws IllegalValueException if there were any data constraints violated.
+     */
+    public NoteCatalogue toModelType() throws IllegalValueException {
+        NoteCatalogue noteCatalogue = new NoteCatalogue();
+        for (JsonAdaptedNote jsonAdaptedNote : notes) {
+            Note note = jsonAdaptedNote.toModelType();
+            if (noteCatalogue.hasNote(note)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_LOCAL_COURSE);
+            }
```
[the remark this answers is not in the document]
Comment (aliciamichellew) [reply src/main/java/seedu/address/storage/JsonSerializableNoteCatalogue.java:57]: Can I ask what do you mean why not reflected on master? I am also not sure of the bug fix you mention here.

[the remark this answers is not in the document]
Comment (aliciamichellew) [reply src/main/java/seedu/address/storage/JsonSerializableNoteCatalogue.java:57]: Seen your PR! Will fix

File: src/main/java/seedu/address/model/notes/Content.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String content;
+
+    /**
+     * Constructs a {@code Content}.
+     *
+     * @param content A valid content.
+     */
+    public Content(String content) {
```
[the remark this answers is not in the document]
Comment (lamchenghou) [reply src/main/java/seedu/address/model/notes/Content.java:26]: Verified.

File: src/main/java/seedu/address/model/notes/UniqueNoteList.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lamchenghou) [reply src/main/java/seedu/address/model/notes/UniqueNoteList.java:26]: Hmm okay, let's leave it for now, I think they are a lot of places that we need to change if we don't use indexing for notes.

**Pull request merged**